lib/auth/saml.go

```diff
@@ -347,17 +349,25 @@ func (a *Server) validateSAMLResponse(samlResponse string) (*samlAuthResponse, e
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	assertionInfo, err := provider.RetrieveAssertionInfo(samlResponse)
+	response, err := provider.ValidateEncodedResponse(samlResponse)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	assertionInfo, err := RetrieveAssertionInfo(provider, connector, response)
```

Comment:
if I'm reading it correctly, `gosaml2` should decrypt assertions automatically as long as `SPKeyStore` is set in the `provider`.
you shouldn't need any changes in this file IIUC.

Reply:
I did try that but it seemed to not under certain conditions. Not an expert on the SAML standard which happens to be very expansive and each IdP does it a little differently. Decided to leave it in in case that happens. Not sure why it happened but to find out I have a lot of digging to do in the gosaml, goxmlsig and etree packages.

Reply:
Can you elaborate on which conditions are causing problems and with which SAML providers?

I'm very worried about re-implementing a part of SAML parsing/validation code correctly, especially when we don't have any SAML experts on the team.

Since there's some time until 6.1, could you please investigate what the `gosaml` library is doing wrong? Once you do, please check upstream. They may have fixed it and we just need a version bump. Or there may be an open PR/issue we can push on.

Reply:
I'll investigate

Reply:
So I did some investigation and I think we've pulled in a newer version of a transitive dependency which fixes it case where it didn't like a certain XML namespace tag in signed & encrypted assertions. I also consulted a friend that knows SAML a lot better than I do and he said that the SAML test data I had in the beginning was technically valid but no one really bothered to support it. IdP's work, it was mostly my test data that was incorrect since I wasn't really sure how to generate it correctly at first.

Reply:
Thanks for getting to the bottom of this!
Does this transitive dependency need to be updated in our `go.mod`?

Reply:
It doesn't, we updated a ton of deps not long ago and I just had to rebase.